src/components/post-unscheduled-check/index.tsx

```diff
@@ -9,18 +9,18 @@ import React, { FC } from 'react';
 import { useSelect } from '@wordpress/data';
 
 export interface PostUnscheduledCheckProps {
-	isFloating: boolean;
+	postStatus: string;
 	hasPublishAction: boolean;
 	isPublished: boolean;
 }
 
 export const PostUnscheduledCheck: FC<PostUnscheduledCheckProps> = ( {
 	children,
 	hasPublishAction,
-	isFloating,
+	postStatus,
 	isPublished,
 } ) => {
-	if ( isPublished || hasPublishAction || ! isFloating ) {
+	if ( isPublished || hasPublishAction || ! [ 'draft', 'future' ].includes( postStatus ) ) {
```

Comment:
This should include `auto-draft` as well, or else the publish time dialogue won't appear for newly-created posts

Reply:
(resolved)